### PR TITLE
feat: add support optional next mask

### PR DESCRIPTION
**Pull request by dima-bzz**
Add support optional next mask.

Usage:

`<input name="ip_address" v-facade="##?#?.##?#?.##?#?.##?#?" />`

### CODE REVIEW

**Comment by dima-bzz** [comment]
How do you think the mask should work?

**Comment by RonaldJerez** [comment]
So looking at your code, you are setting a boolean optional to true, increasing the maskIndex and continuing the loop.

If you wanted to try having the the "?" after, you can set your boolean true, dont increase the maskIndex and letting the loop execute.   Your else if can then become a regular `if (optional)` that will increase the mask by 2 since we need to skip the question mark.

**Comment by dima-bzz** [comment]
Like this?
```
 if (masker.optional) {
    optional = true
    // maskIndex++
    //continue
  }

  if (masker.pattern.test(char)) {
    char = masker.transform ? masker.transform(char) : char
    output.unmasked += char
    output.masked += accumulator + char

    accumulator = ''
    maskIndex++
    optional = false
  }

  if (optional) {
    maskIndex++
    continue
  }
```
This mask `##?#?FFF.###` can be [1FFF.111, 11FFF.111, 111FFF.111]

**Comment by dima-bzz** [comment]
Or don't use the "?", but instead use the optional along with the pattern, like this:
```
   tokens: {
        9: { pattern: /\d/, optional: true },
        '#': { pattern: /\d/ },
        X: { pattern: /[0-9a-z]/i },
        S: { pattern: /[a-z]/i },
        A: { pattern: /[a-z]/i, transform: (v) => v.toLocaleUpperCase() },
        a: { pattern: /[a-z]/i, transform: (v) => v.toLocaleLowerCase() },
        '\\': { escape: true }
    }
```
and

`#99FFF.###`

**Comment by RonaldJerez** [comment]
I think the example using the ? Is better otherwise we would need to add optional for multiple tokens.

I think the code you showed should be ok, but you will need to check the string after the current ask to check if it’s optional l

File: src/masker.js [hunk 1]
```diff
@@ -143,4 +159,4 @@ export default function masker(value, config) {
   return Array.isArray(config.mask)
     ? dynamic(value, Object.assign({}, config, { masks: config.mask }))
     : formatter(value, config)
-}
+}
```
Comment (RonaldJerez) [inline src/masker.js:162]: this is throwing a warning, if you run `npm run lint` it should fix it.

**Pull request merged**